src/pages/groupComparison/ClinicalData.tsx

```diff
@@ -255,17 +255,31 @@ export default class ClinicalData extends React.Component<IClinicalDataProps, {}
     });
 
     private readonly groupSampleDataPromise = remoteData({
-        await: () => [this.props.store.sampleKeyToActiveGroups, this.props.store.uidToGroup, this.props.store._originalGroups],
+        await: () => [this.props.store.sampleSet, this.props.store.activeGroups],
         invoke: async () => {
-            const categoryOrder = _.map(this.props.store._originalGroups.result!, group => group.nameWithOrdinal);
+            const categoryOrder = _.map(this.props.store.activeGroups.result!, group => group.nameWithOrdinal);
             const axisData = { data: [], datatype: "string", categoryOrder } as IAxisData;
-            const uidToGroup = this.props.store.uidToGroup.result!;
+            const uidToGroup = _.keyBy(this.props.store.activeGroups.result!, group => group.uid);
+
+            const sampleKeyToGroups: { [sampleKey: string]: string[] } = {};
+            const sampleSet = this.props.store.sampleSet.result!;
+            for (const group of this.props.store.activeGroups.result!) {
+                for (const studyEntry of group.studies) {
+                    for (const sampleId of studyEntry.samples) {
```

Comment:
this is duplicated logic from groupcomparisonstore? should it be in util function?

Reply:
yup, both are same. I'll get rid off this